Tasks/AzureSpringCloudV0/deploymentProvider/azure-arm-spring-cloud.ts

```diff
@@ -85,29 +79,132 @@ export class AzureSpringCloud {
+        tl.debug('Starting deployment with build service');
+        try {
+            const deploymentTarget = await this.getUploadTargetWithBuildService();
+            await uploadFileToSasUrl(deploymentTarget.sasUrl, artifactToUpload);
+
+            const triggeredBuildResult = await this.updateKPackBuild(appName, deploymentTarget.relativePath, builder);
+            await this.waitBuildFinish(triggeredBuildResult);
+
+            const deploymentUpdateRequestBody = this.prepareDeploymentUpdateRequestBody(deploymentTarget.relativePath, sourceType, runtime, jvmOptions, environmentVariables, dotNetCoreMainEntryPath, version, triggeredBuildResult);
+            await this.applyDeploymentModifications(appName, deploymentName, deploymentUpdateRequestBody, createDeployment);
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    protected async getUploadTargetWithBuildService(): Promise<UploadTarget> {
+        tl.debug('Obtaining upload target with build service');
+
+        const buildServiceName = 'default'
+        const requestUri = this._client.getRequestUri(`${this._resourceId}/buildServices/{buildServiceName}/getResourceUploadUrl`, {
+            '{buildServiceName}': buildServiceName
+        }, null, API_VERSION);
+
+        const response = await this.sendRequest('POST', requestUri, null);
+
+        if (response.statusCode != 200) {
+            console.error('Error code: ' + response.statusCode);
+            console.error(response.statusMessage);
+            throw ToError(response);
+        }
+
+        return new UploadTarget(response.body.uploadUrl, response.body.relativePath);
+    }
+
+    private async updateKPackBuild(appName: string, relativePath: string, builder?: string) {
+        tl.debug('Updating KPack build');
+
+        // Prepare request uri
+        // YITAOPANTODO Is buildServiceName alway to be 'default'
+        // YITAOPANTODO What is buildName 
+        const buildServiceName = 'default';
+        const requestUri = this._client.getRequestUri(`${this._resourceId}/buildServices/{buildServiceName}/builds/{buildName}`, {
+            '{buildServiceName}': buildServiceName,
+            '{buildName}': appName
+        }, null, API_VERSION);
+        // Prepare request body (parameter name - build)
+        const requestBody = {
+            properties: {
+                relativePath: relativePath,
+                builder: builder ? builder : `${this._resourceId}/buildServices/${buildServiceName}/builders/default`,
+                agentPool: `${this._resourceId}/buildServices/default/agentPools/default`,
+                // YITAOPANTODO What is the environment variable in the request body
+                // env = environmentVariables
+            }
+        };
+
+        // Send the request
+        try {
+            const response = await this.sendRequest('PUT', requestUri, JSON.stringify(requestBody));
+            if (response.statusCode == 200) {
```

Comment:
Check with Fenqin whether they always return 200 for success. (Is it possible to have 201/202 kinds of results.)